recipes-bsp/u-boot/files/0020-iot2050-Increase-the-heap-size-for-malloc_simple.patch

```diff
@@ -0,0 +1,44 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Baocheng Su <baocheng.su@siemens.com>
+Date: Tue, 1 Aug 2023 13:46:53 +0800
+Subject: [PATCH] iot2050: Increase the heap size for malloc_simple
+
+The heap is almost full now, it will report `alloc space exhausted` for
+some extended functionality, such as the the RPMB keyring:
+
+| alloc space exhausted
+| Could not get FIT buffer of 1057180 bytes
+|         check CONFIG_SYS_SPL_MALLOC_SIZE
+
+Double the heap size to fix it.
+
+Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
+---
+ configs/iot2050_pg1_defconfig | 1 +
+ configs/iot2050_pg2_defconfig | 1 +
+ 2 files changed, 2 insertions(+)
+
+diff --git a/configs/iot2050_pg1_defconfig b/configs/iot2050_pg1_defconfig
+index 49f811093643ac2cd78b41f0ea50ad6cf4c338f0..c019c890f07ccc9a82e0ef66dd4597985cac4b39 100644
+--- a/configs/iot2050_pg1_defconfig
++++ b/configs/iot2050_pg1_defconfig
+@@ -44,6 +44,7 @@ CONFIG_SPL_SYS_MALLOC_SIMPLE=y
+ CONFIG_SPL_STACK_R=y
+ CONFIG_SYS_SPL_MALLOC=y
+ CONFIG_SYS_SPL_MALLOC_SIZE=0x800000
++CONFIG_SPL_STACK_R_MALLOC_SIMPLE_LEN=0x200000
+ CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR=y
+ CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x1400
+ CONFIG_SPL_DM_MAILBOX=y
+diff --git a/configs/iot2050_pg2_defconfig b/configs/iot2050_pg2_defconfig
+index a31691a1d188ae280ef88311e51e7a5fd909a218..7e231b68ae8bff853b7b4354ff4056dc510bf5eb 100644
+--- a/configs/iot2050_pg2_defconfig
++++ b/configs/iot2050_pg2_defconfig
+@@ -44,6 +44,7 @@ CONFIG_SPL_SYS_MALLOC_SIMPLE=y
+ CONFIG_SPL_STACK_R=y
+ CONFIG_SYS_SPL_MALLOC=y
+ CONFIG_SYS_SPL_MALLOC_SIZE=0x800000
++CONFIG_SPL_STACK_R_MALLOC_SIMPLE_LEN=0x200000
+ CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_SECTOR=y
+ CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR=0x1400
+ CONFIG_SPL_DM_MAILBOX=y
```

Comment:
Please make sure to submit that upstream as well (will only affect a single config file now).

Reply:
Seems TI was bite by the same issue, and they submitted a patch to expand the size by *4.  Use their upstream version instead.